src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -204,6 +204,11 @@ static constexpr auto BLOOM_FILTER_MASK = BLOOM_FILTER_SIZE - 1;
 // constructor does not do what you would expect it to, so try to avoid it.
 using BloomFilter = boost::dynamic_bitset<>;
 
+// EMPTY_BLOOM_FILTER is initialized by creating a BloomFilter with every value being false and using bitwise
+// negation (~x). As the negation is surprisingly expensive, we create a static empty bloom filter and reference
```

Comment:
One more thing. If negation is surprisingly expensive. Why not create an integer and use that one for the initialization using the second constructor?
```
dynamic_bitset(size_type num_bits,
               unsigned long value = 0,
               const Allocator& alloc = Allocator())
```

Could that be faster?

Reply:
"Note that the dynamic_bitset(n, value) constructor does not do what you would expect it to, so try to avoid it."

I don't even remember what it does, but it was really surprising.